src/Bicep.LangServer/Handlers/BicepDidChangeWatchedFilesHandler.cs

```diff
@@ -1,30 +1,46 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
+using System.Collections.Generic;
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
-using Bicep.Core.FileSystem;
+using Bicep.Core;
+using Bicep.Core.Workspaces;
 using Bicep.LanguageServer.CompilationManager;
+using Bicep.LanguageServer.Configuration;
 using MediatR;
 using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
 using OmniSharp.Extensions.LanguageServer.Protocol.Models;
 using OmniSharp.Extensions.LanguageServer.Protocol.Workspace;
+using FileSystemWatcher = OmniSharp.Extensions.LanguageServer.Protocol.Models.FileSystemWatcher;
 
 namespace Bicep.LanguageServer.Handlers
 {
     public sealed class BicepDidChangeWatchedFilesHandler : DidChangeWatchedFilesHandlerBase
     {
+        private readonly IBicepConfigChangeHandler bicepConfigChangeHandler;
+        private readonly IWorkspace workspace;
         private readonly ICompilationManager compilationManager;
-        private readonly IFileResolver fileResolver;
 
-        public BicepDidChangeWatchedFilesHandler(ICompilationManager compilationManager, IFileResolver fileResolver)
+        public BicepDidChangeWatchedFilesHandler(IBicepConfigChangeHandler bicepConfigChangeHandler, ICompilationManager compilationManager, IWorkspace workspace)
         {
+            this.bicepConfigChangeHandler = bicepConfigChangeHandler;
             this.compilationManager = compilationManager;
-            this.fileResolver = fileResolver;
+            this.workspace = workspace;
         }
 
         public override Task<Unit> Handle(DidChangeWatchedFilesParams request, CancellationToken cancellationToken)
         {
-            compilationManager.HandleFileChanges(request.Changes);
+            Container<FileEvent> fileEvents = request.Changes;
+            IEnumerable<FileEvent> bicepConfigFileChangeEvents = fileEvents.Where(x => x.Uri.Path.EndsWith(LanguageConstants.BicepConfigSettingsFileName));
+
+            // Retrigger compilation of source files in workspace when local bicepconfig.json file is edited
+            if (bicepConfigFileChangeEvents.Any())
```

Comment:
I think it's OK to assume that any config file change => recompile everything, but is it worth trying to be more precise (e.g. figuring out which files will actually be impacted by a given config file)?

Might not be worth optimizing for a relatively rare scenario, just wanted to raise it.

Reply:
I made some changes as part of below commit, which should cover this scenario.  Recompilation will only be triggered when LanguageId is "bicep":
https://github.com/Azure/bicep/commit/7c53107f3fb09a3e6403cf76b8ee6a876c13806c

Let me know if I am missing something.